SourceX/miniwin/misc.cpp

```diff
@@ -347,19 +372,21 @@ HWND CreateWindowExA(
 		flags |= SDL_WINDOW_FULLSCREEN;
 	}
 
-	int grabInput = 1;
-	DvlIntSetting("grab input", &grabInput);
 	if (grabInput) {
 		flags |= SDL_WINDOW_INPUT_GRABBED;
 	}
 
 	window = SDL_CreateWindow(lpWindowName, SDL_WINDOWPOS_UNDEFINED, SDL_WINDOWPOS_UNDEFINED, nWidth, nHeight, flags);
+#endif
 	if (window == NULL) {
 		SDL_Log(SDL_GetError());
 	}
 	atexit(FakeWMDestroy);
 
 	if (upscale) {
+#ifdef USE_SDL1
+		SDL_Log("upscaling not supported with USE_SDL1");

```

Comment:
Maybe it would make more sens to force `upscale` to `0` and display the warning in the previous `#ifdef` block, or would you still have to hide everything in the `#else`?

Reply:
Yeah, while we could stub these functions out I'm not sure if the Renderer/Texture approach is the right way to implement upscaling in SDL 1 anyway.

The danger with stubbing is that the code may compile but not actually work with SDL1.

Reply:
Ok leave it unimplementeed in that case. Honestly I don't think we should bother with upscaling for sdl1, full screen should cover cases where sdl1 is relevant.

Reply:
> Ok leave it unimplementeed in that case. Honestly I don't think we should bother with upscaling for sdl1, full screen should cover cases where sdl1 is relevant.

Yeah, in fact in the RetroFW / OpenDingux case the device ships with modified SDL 1.2 so that scaling and fullscreen are actually always done by the library automatically.

It looks like this even though I'm not passing the fullscreen flag and the screen resolution is 320x480 (pixels are not square but line doubling happens somewhere automatically): https://twitter.com/glebm/status/1175785404798316544

Reply:
Nice vid thanks for sharing. We so desperately need to land the game controller code from the switch branch in a clean way 😔